src/Layout.module.css

```diff
@@ -0,0 +1,163 @@
+$spacers: (
+  1: 4px,
+  2: 8px,
+  3: 12px,
+  4: 16px,
+  5: 20px,
+  6: 24px,
+  8: 32px,
+  10: 40px,
+  12: 48px,
+  16: 64px,
+  20: 80px,
+  24: 96px,
+  32: 128px,
+  40: 160px,
+  48: 192px,
+  56: 224px,
+  64: 256px,
+);
+
+.block {
+  display: block;
+}
+.block-inline {
+  display: inline-block;
+}
+
+.flex {
+  display: flex;
+}
+
+.flex-inline {
+  display: inline-flex;
+}
+
+.wrap {
+  flex-wrap: wrap;
+}
+
+.column {
+  flex-direction: column;
+  &.reverse {
+    flex-direction: column-reverse;
+  }
+  & > .spacer {
+    /* important needed to override the pad prop and others generally */
+    margin-top: auto !important;
+  }
+}
+
+.row {
+  flex-direction: row;
+  min-width: 0;
+  max-height: 100%;
+
+  &.reverse {
+    flex-direction: row-reverse;
+  }
+  & > .spacer {
+    /* important needed to override the pad prop and others generally */
+    margin-left: auto !important;
+  }
+}
+
+.grow {
+  flex: 1;
+}
+
+$align-items: (
+  center,
+  start,
+  end,
+  flex-start,
+  flex-end,
+  self-start,
+  self-end,
+  stretch,
+  baseline,
+  first-baseline,
+  last-baseline
+);
+
+@each $item in $align-items {
+  .align-#{$item} {
+    align-items: $item;
+  }
+}
+
+$align-content: (
+  center,
+  start,
+  end,
+  flex-start,
+  flex-end,
+  space-between,
+  space-around,
+  space-evenly,
+  stretch,
+  baseline,
+  first-baseline,
+  last-baseline
+);
+
+@each $content in $align-content {
+  .align-#{$content} {
+    align-content: $content;
+  }
+}
+
+$align-self: (center, flex-start, flex-end, stretch);
+
+@each $self in $align-self {
+  .align-#{$self} {
+    align-self: $self;
+  }
+}
+
+$justify-content: (
+  center,
+  left,
+  right,
+  start,
+  end,
+  flex-start,
+  flex-end,
+  space-between,
+  space-around,
+  space-evenly,
+  stretch,
+  baseline,
+  first-baseline,
+  last-baseline
+);
+
+@each $content in $justify-content {
+  .justify-#{$content} {
+    justify-content: $content;
+  }
+}
+
+@each $level, $spacer in $spacers {
+  .pad-#{$level} {
```

Comment:
i'm somewhat confused by this approach with levels here

it seems like – if we're using CSS variables anyway, why not just pass down the padding as a variable of some sort, esp given that we're just calculating the padding as `level * 4`

likewise for things like the align, justify bits...

can't we just do e.g.

```
justify-content: var(--fc-layout-justify-content)
```

and not have to define a bunch of extra classes?

Reply:
or does that lead to undesirable cascade things?

Reply:
Ooo that's a great idea.

Reply:
O I guess the concern there is all `as` components would need to pass through style as well as className. Idk if that's a reasonable assumption